### PR TITLE
Update to sarama v1.10.0

### PR DESCRIPTION
**Pull request by urso**

### CODE REVIEW

File: libbeat/outputs/kafka/config.go [hunk 1]
```diff
@@ -14,34 +14,58 @@ type kafkaConfig struct {
...
 	defaultConfig = kafkaConfig{
-		Hosts:           nil,
-		TLS:             nil,
-		Timeout:         30 * time.Second,
-		Worker:          1,
+		Hosts:   nil,
+		TLS:     nil,
+		Timeout: 30 * time.Second,
+		Worker:  1,
+		Metadata: metaConfig{
```
Comment (ruflin) [inline libbeat/outputs/kafka/config.go]: What will happen if these settings are used with an older version of Kafka?

Comment (urso) [inline libbeat/outputs/kafka/config.go]: sarama lib supports kafka 0.8 - 0.10 . Settings should be valid for all kafka versions. Lib might internally ignore settings if they do not apply to the version being used. AFAIK metadata updates and publisher settings should have same effect for kafka versions.

File: libbeat/outputs/kafka/client.go [hunk 2]
```diff
@@ -108,16 +108,28 @@ func (c *client) AsyncPublishEvents(
 			topic = event["type"].(string)
 		}
 
+		var ts time.Time
+		if c.config.Version.IsAtLeast(sarama.V0_10_0_0) {
```
Comment (ruflin) [inline libbeat/outputs/kafka/client.go]: A code comment would be nice that quickly explains why whis is only possible with >= 0.10

File: libbeat/outputs/kafka/config.go [hunk 3]
```diff
@@ -58,5 +82,13 @@ func (c *kafkaConfig) Validate() error {
 		return fmt.Errorf("compression mode '%v' unknown", c.Compression)
 	}
 
+	if _, ok := kafkaVersions[c.Version]; !ok {
+		return fmt.Errorf("unknown/unsupported kafka version '%v'", c.Version)
+	}
+
+	if c.Username != "" && c.Password == "" {
+		return fmt.Errorf("password must not be empty when SASL user is configured")
```
Comment (ruflin) [inline libbeat/outputs/kafka/config.go]: I would rephrase this to "empty password. Password config option must be set when Username is configured."

File: libbeat/outputs/kafka/kafka.go [hunk 4]
```diff
@@ -51,6 +51,23 @@ var (
 		"gzip":   sarama.CompressionGZIP,
 		"snappy": sarama.CompressionSnappy,
 	}
+
+	kafkaVersions = map[string]sarama.KafkaVersion{
```
Comment (ruflin) [inline libbeat/outputs/kafka/kafka.go]: I was first thinking something is wrong below as some "entries" were missing because the sorting on the right is not based on the version. I would probably sort based on the values on the right side. But that is more my personal preference. Just wanted to note it as I was first confused.

File: libbeat/outputs/kafka/kafka.go [hunk 5]
```diff
@@ -225,6 +250,7 @@ func newKafkaConfig(config *kafkaConfig) (*sarama.Config, error) {
 		retryMax = 1000
 	}
 	k.Producer.Retry.Max = retryMax
+	// TODO: k.Producer.Retry.Backoff = ?
```
Comment (ruflin) [inline libbeat/outputs/kafka/kafka.go]: Should this stay in?

**Comment by ruflin** [comment]
Doesn't this PR also need an update the libbeat config file?

File: libbeat/outputs/kafka/client.go (hunk 2 above)
Comment (urso) [inline libbeat/outputs/kafka/client.go]: done

File: libbeat/outputs/kafka/config.go (hunk 3 above)
Comment (urso) [inline libbeat/outputs/kafka/config.go]: changed to `password must be set when username is configured`

File: libbeat/outputs/kafka/kafka.go (hunk 4 above)
Comment (urso) [inline libbeat/outputs/kafka/kafka.go]: change ordering

File: libbeat/outputs/kafka/kafka.go (hunk 5 above)
Comment (urso) [inline libbeat/outputs/kafka/kafka.go]: yes. It's some samara internal config with default 100ms kinda overlapping with libbeat functionality (not critical, though). Added `TODO` so I will reconsider a parameter maybe another time instead of hiding the fact something might be "incomplete".

**Comment by urso** [comment]
> Doesn't this PR also need an update the libbeat config file?

Ups, yes.

**Comment by urso** [comment]
@ruflin added missing config file updates + added options to docs
